src/page/index.html

```diff
@@ -18,10 +18,388 @@
       integrity="sha384-EVSTQN3/azprG1Anm3QDgpJLIm9Nao0Yz1ztcQTwFspd3yD65VohhpuuCOmLASjC"
       crossorigin="anonymous"
     />
+    <link href="style.css" rel="stylesheet" />
     <title>サイト制作課題</title>
   </head>
   <body>
-    <h1>このファイルにサイトを作っていくよ。</h1>
+    <header>
+      <div class="header-inner">
+        <a href="#"><img class="header-img" src="../img/header.png" /></a>
```

Comment:
```suggestion
        <a href="#"><img class="header-img" src="../../../img/header.png" /></a>
```
おそらくローカルで確認するためにパスを調整されたと思うのですがgithub pages側では画像がある位置は(./)です。(変更したymlファイルの内容はimgディレクトリをpageフォルダにコピーしてindex.htmlのパス「../../../」を「./」に置き換える処理を行っています)
そのためgithub pagesで画像を表示させたい場合はそのまま../../../とすればokです(もしくは変換後の./でもいいです)。
変更した内容がpagesに反映されるのが遅かったりするのでそれが気になったらsrc/imgフォルダを丸ごとsrc/pageフォルダ下にコピーしてあげればpagesと同じパス関係になるのでそれもありだと思います。